packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -256,6 +263,50 @@ const HoverCardContentImpl = React.forwardRef<
   const popperScope = usePopperScope(__scopeHoverCard);
   const ref = React.useRef<HoverCardContentImplElement>(null);
   const composedRefs = useComposedRefs(forwardedRef, ref);
+  const [containTextSelection, setContainTextSelection] = React.useState(false);
+
+  React.useEffect(() => {
+    if (containTextSelection) {
+      const body = document.body;
+
+      // Safari requires prefix
+      originalBodyUserSelect = body.style.userSelect || body.style.webkitUserSelect;
```

Comment:
Do we feel this restoration is necessary?

Reply:
I mean, unlikely that someone makes it anything on body by default, but for completeness I think we should yes.